### PR TITLE
Fix issue with building x509 on Node.js 12

### PR DESCRIPTION
**Pull request by ghaiklor**
I made all the fixes through nan API, so it should be backward compatible with older version of Node.js.

For those who can't wait for merge, I have a fork, where these changes are already in master - https://github.com/ghaiklor/node-x509

also, there is an npm package already published under my scope - `@ghaiklor/x509`.

### CODE REVIEW

**Comment by ghaiklor** [comment]
Yeah, so, turn out that Node.js 12 has the breaking API in itself that is not compatible with nan, I guess.

@Southern any updates when it can be merged? Since it is a breaking change, better to make major update.

File: README.md [hunk 1]
```diff
@@ -1,13 +1,16 @@
 node-x509
 =========
 
-[![Build Status](https://travis-ci.org/Southern/node-x509.svg?branch=master)](https://travis-ci.org/Southern/node-x509)
+_Forked from <https://github.com/Southern/node-x509>._
+_The reason behind this is that original package does not build on Node.js 12._
+
+[![Build Status](https://travis-ci.com/ghaiklor/node-x509.svg?branch=master)](https://travis-ci.com/ghaiklor/node-x509)
```
Comment (codebling) [inline README.md:7]: This should not be part of the PR

**Comment by codebling** [comment]
There are a number of changes that are part of this PR that prevent it from being merged. Specifically all of the "rebranding" as your own package, most importantly the change to all of the Github URLs towards your repo, and the change in version number. 

All of the style changes and the replacement of the .gitignore file are unnecessary changes that add a lot of noise but otherwise the changes look good!

**Comment by ghaiklor** [comment]
@codebling Thanks for your comment, I totally understand that.

If you take a look into first commit https://github.com/Southern/node-x509/pull/82/commits/d86123a0993b33ca8b9ba6a459487328dfb80a9a

You will find the changes without noise. The commit was made from my `master` branch in fork. Everything ok with that there.

Although, the author did not respond, I made my own changes to have a possibility to publish it on npm. Yeah, like rebranding as well, so other people can install it not from the GitHub, but from npm.

UPD: I'm even not sure if this PR should be opened. It's like half a year old and nothing from the author.

**Comment by codebling** [comment]
> You will find the changes without noise. The commit was made from my `master` branch in fork. Everything ok with that there.
> 

Perhaps you could reopen a PR from a new branch with only that commit? 
```
git checkout -b pr 21fd45098705
git push -u origin pr
```
then go to GitHub and open the PR from the "pr" branch you just created

> Although, the author did not respond, I made my own changes to have a possibility to publish it on npm. Yeah, like rebranding as well, so other people can install it not from the GitHub, but from npm.
> UPD: I'm even not sure if this PR should be opened. It's like half a year old and nothing from the author.

I have had my share of this and it's very frustrating. I'm sorry this has happened to you
@Southern @yorkie
